UnitTests/Schemas/IFC4X1/polygonal-face-tessellation/src/polygonal-face-tessellation.cpp

```diff
@@ -0,0 +1,190 @@
+	const auto expected_top_left_front = buw::loadImage4b(dataPath("polygonal-face-tessellation_top_left_front.png").string());
+	const auto expected_top_front_right = buw::loadImage4b(dataPath("polygonal-face-tessellation_top_front_right.png").string());
+	const auto expected_top_left_back = buw::loadImage4b(dataPath("polygonal-face-tessellation_top_left_back.png").string());
+	const auto expected_top_right_back = buw::loadImage4b(dataPath("polygonal-face-tessellation_top_right_back.png").string());
+	const auto expected_back_left_bottom = buw::loadImage4b(dataPath("polygonal-face-tessellation_back_left_bottom.png").string());
+	const auto expected_right_bottom_back = buw::loadImage4b(dataPath("polygonal-face-tessellation_right_bottom_back.png").string());
+
+	// Act (FrontLeftBottom)
+	renderer->setViewDirection(buw::eViewDirection::FrontLeftBottom);
+	buw::Image4b image_front_left_bottom = CaptureImage();
+	// Act (FrontRightBottom)
+	renderer->setViewDirection(buw::eViewDirection::FrontRightBottom);
+	buw::Image4b image_front_right_bottom = CaptureImage();
+	// Act (TopLeftFront)
+	renderer->setViewDirection(buw::eViewDirection::TopLeftFront);
+	buw::Image4b image_top_left_front = CaptureImage();
+	// Act (TopFrontRight)
+	renderer->setViewDirection(buw::eViewDirection::TopFrontRight);
+	buw::Image4b image_top_front_right = CaptureImage();
+	// Act (TopLeftBack)
+	renderer->setViewDirection(buw::eViewDirection::TopLeftBack);
+	buw::Image4b image_top_left_back = CaptureImage();
+	// Act (TopRightBack)
+	renderer->setViewDirection(buw::eViewDirection::TopRightBack);
+	buw::Image4b image_top_right_back = CaptureImage();
+	// Act (BackLeftBottom)
+	renderer->setViewDirection(buw::eViewDirection::BackLeftBottom);
+	buw::Image4b image_back_left_bottom = CaptureImage();
+	// Act (RightBottomBack)
+	renderer->setViewDirection(buw::eViewDirection::RightBottomBack);
+	buw::Image4b image_right_bottom_back = CaptureImage();
+
+	// uncomment following lines to also save the screen shot
+	
+	buw::storeImage(testPath("polygonal-face-tessellation_front_left_bottom.png").string(), image_front_left_bottom);
+	buw::storeImage(testPath("polygonal-face-tessellation_front_right_bottom.png").string(), image_front_right_bottom);
+	buw::storeImage(testPath("polygonal-face-tessellation_top_left_front.png").string(), image_top_left_front);
+	buw::storeImage(testPath("polygonal-face-tessellation_top_front_right.png").string(), image_top_front_right);
+	buw::storeImage(testPath("polygonal-face-tessellation_top_left_back.png").string(), image_top_left_back);
+	buw::storeImage(testPath("polygonal-face-tessellation_top_right_back.png").string(), image_top_right_back);
+	buw::storeImage(testPath("polygonal-face-tessellation_back_left_bottom.png").string(), image_back_left_bottom);
+	buw::storeImage(testPath("polygonal-face-tessellation_right_bottom_back.png").string(), image_right_bottom_back);
+
```

Comment:
In the end version, these lines should be commented out.

Reply:
Resolved 1681521